sycl/test-e2e/Graph/UnsupportedDevice/device_query.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests the using device query for graphs support, and that the return value
+// matches expectations.
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  auto Device = Queue.get_device();
+  bool SupportsGraphs = Device.has(aspect::ext_oneapi_graph);
+  auto Backend = Device.get_backend();
+
+  if ((Backend == backend::ext_oneapi_level_zero) ||
+      (Backend == backend::ext_oneapi_cuda) ||
+      (Backend == backend::ext_oneapi_hip)) {
+    assert(SupportsGraphs);
+  } else if (Backend != backend::opencl) {
```

Comment:
```suggestion
  } else if (Backend == backend::opencl) {
```
This condition seems wrong.

Reply:
The intention was to not test OpenCL, because we need more information to know if the OpenCL implementation will support graphs or not. Testing false would fail with the emulation layer, which we've had bug reports with before, so I'm not that keen to hardcode false.

For now i've checked if the backend supports the "cl_khr_command_buffer" extension to base ext_oneapi_graph support off, but we'll need to refine this logic later when we add more features.

Reply:
After chatting to Ben offline decided to skip this test on OpenCL